lib/kino_explorer/data_transform_cell.ex

```diff
@@ -907,13 +907,19 @@ defmodule KinoExplorer.DataTransformCell do
 
   defp maybe_update_datalist(operation, _df), do: operation
 
-  defp valid_data(%DataFrame{}), do: true
+  defp valid_data(%DataFrame{} = df), do: {DataFrame.dtypes(df), true}
 
   defp valid_data(data) do
     with true <- implements?(Table.Reader, data),
          {_, %{columns: [_ | _] = columns}, _} <- Table.Reader.init(data),
          true <- Enum.all?(columns, &implements?(String.Chars, &1)) do
-      true
+      try do
+        dtypes = DataFrame.new(data) |> DataFrame.dtypes()
```

Comment:
Building this will unfortunately be very expensive. The only thing I can think of is to allow people to choose the var and then we show the error inside the smart cell.

Reply:
@jonatanklosko that will still hide bugs if the columns/rows have mixed types, so it doesn't replace the need for checking later.

Reply:
Ah we also use the type to determine compatible operations.

Reply:
And also to filter supported operations based on types

Reply:
> Building this will unfortunately be very expensive. The only thing I can think of is to allow people to choose the var and then we show the error inside the smart cell.

Okay! I'll close this PR and follow this approach instead